src/components/Bookmark.js

```diff
@@ -1,57 +1,84 @@
+  const localPassKey = props.passingKey; //can be marginNoteKey from CC or provision from MVA
+  const localData = props.data; //needs this data to pass through redux for rendering purposes.
   const dispatch = useDispatch();
+  const isFocused = useIsFocused(); //checks for state change of mark when screen is focussed (required when switching tab navigation components)
+  const bookmarkStateId = useSelector(state => state.bookmarks.bookmarkArray); //retrieves list of current bookmarks
+
+  //reloads the data state anytime new data is moved into the component.
+  useEffect(() => {
+    setFullSomeData(localData);
+  }, [localData]);
+
+  //function to check to see if items exist in the bookmark redux
+  const checkBookmarkArray = () => {
+    const checkBool =
+      bookmarkStateId.some(
+        e => e.lawType === localLawType && e.passingKey === localPassKey, //passingkey: for CC it's marginNoteKey, for MVA it's provision
+      ) && isFocused;
+    return checkBool;
+  };
+
+  // compares state array to see if section exists in bookmarks, if it does turn on bookmark icon
+  //isFocused is used to force the bookmark to refresh when user tabs back to accordion screen.
+  useEffect(() => {
+    if (checkBookmarkArray() === true) {
+      setMarked(true);
+    } else {
+      setMarked(false);
+    }
+  }, [fullSomeData, marked, isFocused]);
 
-  //console.log(section, sectionHeader);
   //switches state of bookmark
   const switchMarks = () => {
     setMarked(!marked);
   };
 
-  const action = (section, sectionHeader) => {
-    if (marked === true) {
+  //calls the redux methods for adding/remove from the redux store.  "marked" is opposite of logic because of the way react-native handles the use effect state changes.
+  const dispatchAction = (bmData, bmKey, lawT) => {
+    if (marked === false) {
       dispatch(
         addBookmark({
-          section: section,
-          sectionHeader: sectionHeader,
+          legislationGroup: bmData, //adds all data for output purposes on BookmarkScreen.js,
+          passingKey: bmKey, //for CC it's marginNoteKey, for MVA it's provision
+          lawType: lawT,
         }),
       );
     }
-    if (marked === false) {
+    if (marked === true) {
       dispatch(
         removeBookmark({
-          section: section,
+          passingKey: bmKey, //for CC it's marginNoteKey, for MVA it's provision
+          lawType: lawT,
         }),
       );
     }
   };
 
-  useEffect(() => {
-    action(section, sectionHeader);
-  });
-
+  //render portion to implement the bookmark icon in the appropriate place.
   return (
     <Pressable
       onPress={() => {
         switchMarks();
+        dispatchAction(fullSomeData, localPassKey, localLawType);
       }}>
       <View>
-        <Icon
-          name={marked ? 'bookmark' : 'bookmark-outline'}
-          size={30}
-          style={{color: colors.primary}}
-        />
+        {marked === true && isFocused ? (
+          <BookmarkMarked />
```

Comment:
Could these names be any more confusing :D

Reply:
They TOTALLY make sense.